### PR TITLE
PSP-289: Update timelog to state variable

### PR DESCRIPTION
**Pull request by jdtoombs**
Previously the time log would go stale and carry over incorrectly to a new snippet/ an existing one. Tied it to a state variable so it correctly resets to it's corresponding formik values when the "page" changes

### CODE REVIEW

File: app/editor/src/features/content/form/PropertiesContentForm.tsx [hunk 1]
```diff
@@ -214,12 +214,15 @@ export const PropertiesContentForm: React.FC<IContentSubForms> = ({ setContent,
...
           onClick={() => {
             setEffort(effort!! + Number((values as any).prep));
-            timeLog.push({
-              userId: userId ?? 0,
-              activity: !!values.id ? 'Updated' : 'Created',
-              effort: (values as any).prep,
-              createdOn: new Date(),
-            });
+            setTimeLog([
+              ...timeLog,
```
Comment (jdtoombs) [inline app/editor/src/features/content/form/PropertiesContentForm.tsx:218]: Didn't know you could do this with state arrays...

File: app/editor/src/features/content/form/PropertiesContentForm.tsx [hunk 2]
```diff
@@ -38,7 +38,7 @@ export const PropertiesContentForm: React.FC<IContentSubForms> = ({ setContent,
   const [effort, setEffort] = React.useState(getTotalTime(values.timeTrackings));
 
   const userId = users.find((u: IUserModel) => u.username === keycloak.getUsername())?.id;
-  let timeLog = values.timeTrackings;
```
Comment (jdtoombs) [inline app/editor/src/features/content/form/PropertiesContentForm.tsx:41]: culprit

File: app/editor/src/features/content/form/PropertiesContentForm.tsx [hunk 3]
```diff
@@ -35,10 +35,13 @@ export const PropertiesContentForm: React.FC<IContentSubForms> = ({ setContent,
   const [categoryOptions, setCategoryOptions] = React.useState<IOptionItem[]>([]);
   const [seriesOptions, setSeriesOptions] = React.useState<IOptionItem[]>([]);
   const [licenseOptions, setLicenseOptions] = React.useState<IOptionItem[]>([]);
-  const [effort, setEffort] = React.useState(getTotalTime(values.timeTrackings));
+  const [effort, setEffort] = React.useState(0);
 
   const userId = users.find((u: IUserModel) => u.username === keycloak.getUsername())?.id;
-  let timeLog = values.timeTrackings;
+
+  React.useEffect(() => {
```
Comment (jdtoombs) [inline app/editor/src/features/content/form/PropertiesContentForm.tsx:42]: Fixes a separate bug

**Review by Fosol — APPROVED** [review]
I didn't have time to look into this, but I thought it was state related.  Thanks for fixing.

**Pull request merged**